units/SDL_mutex.inc

```diff
@@ -0,0 +1,746 @@
+ *
+ * \since This function is available since SDL 3.2.0.
+ *
+ * \sa SDL_SignalCondition
+ * \sa SDL_WaitCondition
+ * \sa SDL_WaitConditionTimeout
+ *}
+procedure SDL_BroadcastCondition(cond: PSDL_Condition); cdecl;
+  external SDL_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_SDL_BroadcastCondition' {$ENDIF} {$ENDIF};
+
+{**
+ * Wait until a condition variable is signaled.
+ *
+ * This function unlocks the specified `mutex` and waits for another thread to
+ * call SDL_SignalCondition() or SDL_BroadcastCondition() on the condition
+ * variable `cond`. Once the condition variable is signaled, the mutex is
+ * re-locked and the function returns.
+ *
+ * The mutex must be locked before calling this function. Locking the mutex
+ * recursively (more than once) is not supported and leads to undefined
+ * behavior.
+ *
+ * This function is the equivalent of calling SDL_WaitConditionTimeout() with
+ * a time length of -1.
+ *
+ * \param cond the condition variable to wait on.
+ * \param mutex the mutex used to coordinate thread access.
+ *
+ * \threadsafety It is safe to call this function from any thread.
+ *
+ * \since This function is available since SDL 3.2.0.
+ *
+ * \sa SDL_BroadcastCondition
+ * \sa SDL_SignalCondition
+ * \sa SDL_WaitConditionTimeout
+ *}
+procedure SDL_WaitCondition(cond: PSDL_Condition; mutex: PSDL_Mutex); cdecl;
+  external SDL_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_SDL_WaitCondition' {$ENDIF} {$ENDIF};
+
+{**
+ * Wait until a condition variable is signaled or a certain time has passed.
+ *
+ * This function unlocks the specified `mutex` and waits for another thread to
+ * call SDL_SignalCondition() or SDL_BroadcastCondition() on the condition
+ * variable `cond`, or for the specified time to elapse. Once the condition
+ * variable is signaled or the time elapsed, the mutex is re-locked and the
+ * function returns.
+ *
+ * The mutex must be locked before calling this function. Locking the mutex
+ * recursively (more than once) is not supported and leads to undefined
+ * behavior.
+ *
+ * \param cond the condition variable to wait on.
+ * \param mutex the mutex used to coordinate thread access.
+ * \param timeoutMS the maximum time to wait, in milliseconds, or -1 to wait
+ *                  indefinitely.
+ * \returns true if the condition variable is signaled, false if the condition
+ *          is not signaled in the allotted time.
+ *
+ * \threadsafety It is safe to call this function from any thread.
+ *
+ * \since This function is available since SDL 3.2.0.
+ *
+ * \sa SDL_BroadcastCondition
+ * \sa SDL_SignalCondition
+ * \sa SDL_WaitCondition
+ *}
+function SDL_WaitConditionTimeout(
+  cond: PSDL_Condition;
+  mutex: PSDL_Mutex;
+  timeoutMS: cint32
+): cbool; cdecl;
+  external SDL_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_SDL_WaitConditionTimeout' {$ENDIF} {$ENDIF};
+
+{**
+ * The current status of an SDL_InitState structure.
+ *
+ * \since This enum is available since SDL 3.2.0.
+ *}
+type
```

Comment:
This enum should be translated according to our translation rules. (typed constants)